library/core/src/mem/maybe_uninit.rs

```diff
@@ -314,8 +314,10 @@ impl<T> MaybeUninit<T> {
     /// let data = read(&mut buf);
     /// ```
     #[unstable(feature = "maybe_uninit_uninit_array", issue = "none")]
+    #[rustc_const_unstable(feature = "const_maybe_assume_init", issue = "none")]
+    #[rustc_allow_const_fn_unstable(const_maybe_assume_init)]
```

Comment:
Is `rustc_allow_const_fn_unstable` truly needed here? I thought it is only needed for const-stable functions?

Reply:
Also, I think it'd make more sense to make the const stability part of the maybe_uninit_uninit_array feature gate -- the fact that `assume_init` is used is an implementation detail.